Add XML file upload feature for editor

techytushar opened the pull request:
* XML file upload using textarea, fileupload, license name, new document.
* XML beautify feature.
* Write tests

techytushar commented:
@rtgdk @goneall This PR contains the code for the page where the user can upload XML file.

I have used sessions to implement this, whenever user give XML input on the `app/xml_upload` page, a unique random id is generated and he gets redirected to the `app/editor/<unique_id>` page. All the XML data gets stored in the session variables. This ensures that user edit multiple XML files in different browser tabs and data does not get lost even when user refresh the page.

I have built the page on top of base template, so the UI remains the same. Please use it and let me know if some features are not working properly or you want anything to work differently.   

I have written tests for all the possibilities that I could think of, please suggest if some more tests can be added.

Next I will will start working  on the editor, for the editor I have made a basic template, will improve on it. :slightly_smiling_face:

rtgdk commented:
@techytushar  Excellent work there! I really like the license name search with requests. The tests look complete for now. You can add more once you progress with the application. 
Also, about storing the data in sessions. That's a good idea. But I think you should let the user know that the url created is accessible only till he quits the browser. So it's better for them to download it before quitting. 
Also, I would like to merge this work with `xml-editor` branch rather than the master branch until the editor is in working phase.

@goneall  Please review the editor and provide your comments/changes.

techytushar commented:
yes, an alert will be generated whenever the user exits the page, will add it in upcoming commits.

techytushar commented on src/app/views.py:
```diff
@@ -611,6 +614,235 @@ def check_license(request):
     else:
         return HttpResponseRedirect(settings.LOGIN_URL)
 
+def xml_upload(request):
```
@goneall I didn't understand what you are trying to say here, should I change the function name to something more appropriate? Please explain. :sweat_smile:

techytushar replied:
Thank you @goneall for such detailed explanation, I didn't know about code refactoring before.

techytushar commented:
I have added logging, changed the status codes and return messages and now the license identifier is being extracted from the json file. Also modified the tests accordingly. Please review.

techytushar commented:
@goneall @rtgdk I have added the basic XML text editor. I have used CodeMirror editor to implement this which uses the MIT License and made it to work for SPDX License Schema. It includes suggestion based on the schema, autoclose tags, code folding, highlighting matching tags and trailing spaces, fullscreen editing.
The file download feature is also working now, but there are still some bugs in the XML beautify script which will be fixed in upcoming commits.

Please review the changes and suggest edits.

goneall commented:
Can we change this pull request to be against the XML-Editor branch?  @rtgdk once you've reviewed, if you're OK - we can merge it in.

goneall commented:
I would like publish a version to the internet for the legal team to review and comment on.  @techytushar  - let me know when you think there is a version stable enough for user review.

techytushar commented:
Sure, a stable version of the text based editor should be ready by Monday.

rtgdk commented:
@goneall  Work looks good to me. 
@techytushar You are going with nice speed. I would love to see the stable version.
For now, I'll merge this with the xml-editor branch. You can open a new PR to keep things simple. You are also encouraged to update the wiki with the new features you created, or add something in the Todo list if you plan to do it in the later stage. The wiki page will be useful to make people understand how things works and also at the end of GSoC you are required to submit a link describing your project, so wiki will be very useful having all the things in one place.

techytushar commented:
Thank you for merging @rtgdk , I'll create a page in the Wiki :slightly_smiling_face:
